### PR TITLE
Used more specific exceptions in IRSA #1037

### PR DESCRIPTION
**Pull request by tinumide**

### CODE REVIEW

File: astroquery/irsa/core.py [hunk 1]
```diff
@@ -111,7 +111,11 @@
 from ..query import BaseQuery
 from ..utils import commons
 from . import conf
-from ..exceptions import TableParseError, NoResultsWarning
+from ..exceptions import (
```
Comment (bsipocz) [inline astroquery/irsa/core.py:114]: no need for the linebreak here, we allow longer than 80 char lines, up to 100-120 is fine.

File: astroquery/irsa/core.py [hunk 2]
```diff
@@ -368,24 +372,25 @@ def _parse_result(self, response, verbose=False):
...
         # Check to see that the query engine is working
         if 'SQLConnect failed' in content:
-            raise Exception("The IRSA server is currently down")
+            raise TimeoutError("The IRSA server is currently down")
 
         # Check that the results are not of length zero
         if len(content) == 0:
-            raise Exception("The IRSA server sent back an empty reply")
+            warnings.warn("The IRSA server sent back an empty reply",
+                          NoResultsWarning)
```
Comment (bsipocz) [inline astroquery/irsa/core.py:393]: This is great to change this to the warning, however, an empty table has to be returned now after the warning.

**Review by bsipocz — COMMENTED** [review]
This looks very good. I left some code reviews, and since strictly speaking this is an API change, a changelog entry would be great, too.
Something along the lines of "Using more specific exceptions." is enough, along with the number of the PR.
Thanks!

File: astroquery/irsa/core.py (hunk 2 above)
Comment (tinumide) [reply astroquery/irsa/core.py:393]: Thanks @bsipocz. No warning class seems to describe this case perfectly, should I add a `EmptyResultWarning` class in the **exceptions file** that can take care of situations like this ?

File: astroquery/irsa/core.py (hunk 1 above)
Comment (tinumide) [reply astroquery/irsa/core.py:114]: Great, I'll adjust this.

File: astroquery/irsa/core.py (hunk 2 above)
Comment (tinumide) [reply astroquery/irsa/core.py:390]: Also, the check for an empty table is done here https://github.com/tinumide/astroquery/blob/22a6f09b0381fae187f63de18813478d436d0508/astroquery/irsa/core.py#L407 and an empty table is returned afterwards. Should I still go ahead to return the empty table here as well ?

Comment (bsipocz) [reply astroquery/irsa/core.py:390]: No, I think the `NoResultsWarning` is perfectly suitable here, we are using it for similar situations.

And you're absolutely right, the votable machinery already takes care of handling the table parsing, so no need to manually return an empty table.

**Review by bsipocz — APPROVED** [review]
All review comments are addressed

**Comment by bsipocz** [comment]
Thank you @tinumide and welcome to astroquery! 🎉

**Comment by tinumide** [comment]
Yaaaay! 🥳🥳🥳

**Pull request merged**